pkg/strategy/xgap/strategy.go

```diff
@@ -323,34 +326,59 @@ func buildAdjustPositionOrder(
 	return
 }
 
+func (s *Strategy) tryPlaceAdjustPositionOrder(ctx context.Context, bestBid, bestAsk types.PriceVolume, balances types.BalanceMap) bool {
+	positionSnapshot := getPositionSnapshot(s.Position)
+	s.logger.Infof("position snapshot: %+v", positionSnapshot)
+	adjPosIsPossible, adjPosOrder := buildAdjustPositionOrder(
+		s.Symbol,
+		positionSnapshot,
+		bestBid,
+		bestAsk,
+	)
+	if !adjPosIsPossible {
+		return false
+	}
+	// sanity checks on the order
+	// check dust quantity
+	if s.tradingMarket.IsDustQuantity(adjPosOrder.Quantity, adjPosOrder.Price) {
+		s.logger.Warnf("adjust position order is of dust quantity: %+v", adjPosOrder)
+		return false
+	}
+	// check balance
+	balanceOk, balanceErr := tradingutil.HasSufficientBalance(
```

Comment:
Don't we have to check the balance error? Or it implies if balanceOk is false, the error will always exist?

Reply:
I think what I try to achieve here is to make placing order more robust.
In addition to that, I add some logging suggesting insufficient balance which can help debugging in production.